`normalize-request` function `url` is manipulating the url and breaking the request

## Description

`@pollyjs/core/src/utils/normalize-request` function `url` is manipulating the url and breaking the request.

When query parameter does not have a value, the function `url` appends a `=`  before making the request. Specifically consuming a WSDL endpoint for a SOAP client does not work because the request url is changed from `?WSDL` to be `?WSDL=`

### Shareable Source

```
index 029eb01..e8e94a3 100644
--- a/packages/@pollyjs/core/tests/unit/utils/normalize-request-test.js
+++ b/packages/@pollyjs/core/tests/unit/utils/normalize-request-test.js
@@ -124,6 +124,10 @@ describe('Unit | Utils | Normalize Request', function() {
       );
     });
 
+    it('should allow WSDL param', function() {
+      expect(url('http://foo.com?WSDL', {})).to.equal('http://foo.com?WSDL');
+    });
+
     it('should respect `matchRequestsBy.url` config', function() {
       [
         [
```

### Error Message & Stack Trace

```
  1) @pollyjs/core
       Unit | Utils | Normalize Request
         url
           should allow WSDL param:

      AssertionError: expected 'http://foo.com?WSDL=' to equal 'http://foo.com?WSDL'
      + expected - actual

      -http://foo.com?WSDL=
      +http://foo.com?WSDL
      
      at Context.<anonymous> (packages/@pollyjs/core/build/node/test-bundle.cjs.js:8305:49)
      at processImmediate (internal/timers.js:439:21)



error Command failed with exit code 1.
```

### Config
### Dependencies

Was using the source from this repo @ `v4.0.1`
https://github.com/Netflix/pollyjs/tree/v4.0.1

## Relevant Links

In this instance, I was trying to use https://www.npmjs.com/package/soap

## Environment

```
darwin 18.7.0
Node.js v12.14.1
yarn 1.21.1
```

Trying to help by narrowing things down a bit.

`@pollyjs/utils/src/utils/url` is where the undesirable manipulation happens.

It seems the second argument `parse` is critical
https://github.com/Netflix/pollyjs/blob/ed8fe999f1facd7849f389a26da4ea27f36969a4/packages/%40pollyjs/utils/src/utils/url.js#L61-L66

You can see here that `parse=true` in `normalize-request`
https://github.com/Netflix/pollyjs/blob/ed8fe999f1facd7849f389a26da4ea27f36969a4/packages/%40pollyjs/core/src/utils/normalize-request.js#L20

### Shareable Source

```
diff --git a/packages/@pollyjs/utils/tests/unit/utils/url-test.js b/packages/@pollyjs/utils/tests/unit/utils/url-test.js
index 0f16293..fb4e33a 100644
--- a/packages/@pollyjs/utils/tests/unit/utils/url-test.js
+++ b/packages/@pollyjs/utils/tests/unit/utils/url-test.js
@@ -70,4 +70,8 @@ describe('Unit | Utils | URL', function() {
       expect(decode(url.toString().split('?')[1])).to.equal(query);
     });
   });
+
+  it('should work with WSDL', function() {
+    expect(new URL('http://foo.com?WSDL', true).href).to.equal('http://foo.com?WSDL');
+  });
 });
```

### Error Message & Stack Trace
```
  1) @pollyjs/utils
       Unit | Utils | URL
         should work with WSDL:

      AssertionError: expected 'http://foo.com?WSDL=' to equal 'http://foo.com?WSDL'
      + expected - actual

      -http://foo.com?WSDL=
      +http://foo.com?WSDL
      
      at Context.<anonymous> (packages/@pollyjs/utils/build/node/test-bundle.cjs.js:2056:58)
      at processImmediate (internal/timers.js:439:21)



error Command failed with exit code 1.
```

--- a/packages/@pollyjs/utils/src/utils/url.js
+++ b/packages/@pollyjs/utils/src/utils/url.js
@@ -9,12 +9,17 @@ function parseQuery(query, options) {
   return qs.parse(query, {
     plainObjects: true,
     ignoreQueryPrefix: true,
+    strictNullHandling: true,
     ...options
   });
 }
 
 function stringifyQuery(obj, options = {}) {
-  return qs.stringify(obj, { addQueryPrefix: true, ...options });
+  return qs.stringify(obj, {
+    addQueryPrefix: true,
+    strictNullHandling: true,
+    ...options
+  });
 }
 
 /**

--- a/packages/@pollyjs/utils/tests/unit/utils/url-test.js
+++ b/packages/@pollyjs/utils/tests/unit/utils/url-test.js
@@ -19,6 +19,7 @@ describe('Unit | Utils | URL', function() {
   it('should correctly parse query params', function() {
     [
       ['', {}],
+      ['a&b=', { a: null, b: '' }],
       ['foo=bar', { foo: 'bar' }],
       ['a[]=1&a[]=2', { a: ['1', '2'] }],
       ['a[1]=1&a[0]=2', { a: ['2', '1'] }],
@@ -33,6 +34,7 @@ describe('Unit | Utils | URL', function() {
     [
       // Query string will be undefined but we decode it in the assertion
       [{}, decode(undefined)],
+      [{ a: null, b: '' }, 'a&b='],
       [{ foo: 'bar' }, 'foo=bar'],
       [{ a: ['1', '2'] }, 'a[0]=1&a[1]=2'],
       [{ foo: { bar: { baz: '1' } } }, 'foo[bar][baz]=1']

